open-apparel-registry/open-apparel-registry
# Approval workflow updates

## Overview

This PR addresses two isses with the approval workflow:

1. When a facility list is rejected, that facility list's source remains active.
2. Facility lists that are replaced by another facility list remain in the Pending status. A new meta-status, Replaced, was created for these lists.

Connects #2326 

## Notes

Instead of creating a new status for Replaced lists, we had the option to create a "real" status. This would have been simpler to implement, but could lead to data inconsistency (e.g. a non-replaced list with the Replaced status). In addition, if a replacing list is rejected and we want to return the original list, there would be no way to know its original status. 

## Testing Instructions

- `./scripts/test`
---
- Upload a facility list
- `./scripts/manage batch_process --list-id <uploaded list ID> --action parse`
- Login as an admin of the uploaded list's contributor (must be a different user)
- Reject uploaded, pending list
  - [x] Ensure list is Rejected status
  - [x] Ensure list's source is not active

- Upload a facility list
- Upload a replacement list
  - [x] Ensure status of new list is Replaced
  - [x] Ensure it doesn't show up in the UI facility list lists
    - [x] Except for the admin dashboard list when 'Rejected' is selected


## Checklist
